launcher/SysInfo.cpp

```diff
@@ -51,18 +98,58 @@ QString useQTForArch()
     return QSysInfo::currentCpuArchitecture();
 }
 
+uint64_t getSystemRamMiB()
+{
+#if defined(Q_OS_WINDOWS)
+    MEMORYSTATUSEX status;
+    status.dwLength = sizeof status;
+
+    if (GlobalMemoryStatusEx(&status)) {
+        // transforming bytes -> mib
+        return (uint64_t)status.ullTotalPhys / 1024 / 1024;
+    }
+#elif defined(Q_OS_LINUX)
+    struct sysinfo info;
+
+    if (sysinfo(&info) != -1) {
+        // transforming bytes -> mib
+        return info.totalram / 1024 / 1024;
+    }
+#elif defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
+    char buff[512];
+    FILE* fp = popen("sysctl hw.physmem", "r");
+    if (fp != nullptr) {
+        if (fgets(buff, 512, fp) != nullptr) {
+            std::string str(buff);
+            uint64_t mem = std::stoull(str.substr(12, std::string::npos));
+
+            // transforming kib -> mib
+            return mem / 1024;
+        }
+    }
+#elif defined(Q_OS_APPLE)
+    uint64_t memsize;
+    size_t memsizesize = sizeof memsize;
+
+    if (!sysctlbyname("hw.memsize", &memsize, &memsizesize, nullptr, 0)) {
+        // transforming bytes -> mib
+        return memsize / 1024 / 1024;
+    }
```

Comment:
I think we can combine all 3 of these by only slightly modifying the Mac branch (the fBSD and macOS could also be shared as is...but OpenBSD doesn't seem to have `sysctlbyname` so)

```suggestion
// https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/sysctl.3.html
// https://man.freebsd.org/cgi/man.cgi?sysctl(3)
// https://man.openbsd.org/sysctl.2
#elif defined(Q_OS_APPLE) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
    int sysctlVector[2];
    sysctlVector[0] = CTL_HW;
    // `HW_PHYSMEM` is deprecated in OpenBSD
    #if defined(Q_OS_OPENBSD)
    sysctlVector[1] = HW_PHYSMEM64;
    #else
    sysctlVector[1] = HW_PHYSMEM;
    #endif
    uint64_t memsize;
    size_t memsizesize = sizeof memsize;

    if (!sysctl(sysctlVector, &memsize, &memsizesize, nullptr, 0)) {
        // transforming bytes -> mib
        return memsize / 1024 / 1024;
    }
```

Reply:
As long as someone can test it